dao/src/main/java/se/fortnox/reactivewizard/db/DbProxy.java

```diff
@@ -123,7 +129,24 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Throwabl
                     converterFromFlux(method.getReturnType()));
             statementFactories.put(method, reactiveStatementFactory);
         }
-        return reactiveStatementFactory.create(args, connectionScheduler);
+
+        if (isFluxOrMono(method.getReturnType())) {
```

Comment:
Avoid reflection at runtime, this check can be done inside the block above, which will only happen once per method.